### ISSUE TITLE
Improve the collision detection for the Dino and Cactus

### ISSUE THREAD

**Issue by vikram-kangotra** [vikram-kangotra/Dino#1]
Right now, the collision takes place by taking only the bounding boxes (of dino and cactus) into consideration. Improve this to make them collide only when the bodies collide.

**Comment by vidhan63-soe** [comment]
I get it, let me handle this.

**Comment by vikram-kangotra** [comment]
Yeah sure :)

**Comment by chris--jones** [comment]
@vidhan63-soe - I was curious on this one too, how are you planning on solving this?

Two options I considered were overlapping dark pixels or polygon intersection.

**Comment by vikram-kangotra** [comment]
Polygon intersection, while powerful and versatile, might be a bit of an overkill for a game with less complex shapes and interactions.

So, in the context of a simple dino game, I'd lean towards the option of overlapping dark pixels. It aligns with the simplicity of the game and should serve our purpose efficiently.

**Comment by vikram-kangotra** [comment]
Hi @vidhan63-soe, 

Are you working on this?

**Comment by vidhan63-soe** [comment]
yaa i am, but i am stuck on logical error -"Dino is not crossing a single obstacle".

**Comment by vikram-kangotra** [comment]
Were you able to resolve it? 
If not, feel free to submit a merge request in draft mode, allowing others to view the code and provide their feedback.

**Comment by chris--jones** [comment]
A potentially simpler option would be to approximate using multiple bounding boxes, e.g.
![image](https://github.com/vikram-kangotra/Dino/assets/10459377/c92bbcbe-b790-402b-88c7-58d22582d41e)

This way the code isn't too dissimilar to what is there now (you simply need to check all boxes for collisions).

Obviously it wouldn't be as accurate, but it might be a nice iteration to improve things until someone implements pixel comparison or a proper polygon.

Also if you wanted to be super lazy, you could just exclude a bottom right triangle like so:
![image](https://github.com/vikram-kangotra/Dino/assets/10459377/97850146-1cec-4dca-95ec-b90ee28a0ae5)

Basically you can add a tolerance to the collisions depending on the relative positions - so the right collision has no tolerance when we aren't jumping because of the dino's head, but as we move up (jumping) we collide less until we've completely cleared the cactus or hit the body if we're too slow to react.

**Comment by vikram-kangotra** [comment]
Hi @chris--jones @vidhan63-soe 

I have added PR #39 for fixing this issue.
There are some known issues. If you can work on those, it would be a great help :)
